tests/integration/istioio/examples/scripts/bookinfo.txt

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+REPEAT=${REPEAT:-100}
+THRESHOLD=${THRESHOLD:-20}
+
+function verify_services {
+  unmatchedText=""
+  wantTextPattern=("$@")
+  goodResponse=0
+
+  for ((i=1; i<="$REPEAT"; i++)); do
+    tmpfile=$(mktemp /tmp/verify-services.XXXXXX)
+    set +e
+    $(kubectl get services > "${tmpfile}")
+    set -e
+    body=$(cat ${tmpfile})
+
+    matchedText=0
+    for want in "${wantTextPattern[@]}"; do
+       if [ "`grep "${want}" "${tmpfile}"`" != "" ]; then
+        matchedText=$((matchedText + 1))
+      else
+        unmatchedText="${want}"
+      fi
+    done
+
+    if [[ "$matchedText" == "$#" ]]; then
+      goodResponse=$((goodResponse + 1))
+    else
+      goodResponse=0
+    fi
+
+    if (( "$goodResponse">="$THRESHOLD" )); then
+      return 0
+    fi
+  done
+  rm "$tmpfile"
+
+  echo -e "did not match text: ${unmatchedText} in ${body}\n"
+  return 1
+}
+
+function verify_pods {
+  unmatchedText=""
+  wantTextPattern=("$@")
+  goodResponse=0
+
+  for ((i=1; i<="$REPEAT"; i++)); do
+    tmpfile=$(mktemp /tmp/verify-pods.XXXXXX)
```

Comment:
could we do this without creating a file?

I think it's worth nailing this down as a function now. I'd like to build this into the framework.

Reply:
regex matching via grep (or grep -E) seems more intuitive against file (could ideally match beginning/end of lines).